File: src/lib/api/decision.ts
import { fetchApi } from './client';

export interface AuditLogEntry {
  log_id: string;
  timestamp: string;
  event_type: string;
  actor: string;
  details: string;
}

export interface PlannerOverrideRequest {
  planner_name: string;
  planner_role: string;
  reason: string;
  new_start_time: string;
  new_duration_min: number;
}

export async function getAuditLog() {
  return fetchApi<AuditLogEntry[]>('/decision/audit-log');
}

export async function submitPlannerOverride(request: PlannerOverrideRequest) {
  return fetchApi<{ status: string; audit_entry: AuditLogEntry }>('/decision/override', {
    method: 'POST',
    body: JSON.stringify(request),
  });
}

File: backend/app/api/routes/decision.py
from fastapi import APIRouter
from typing import List, Dict, Any
from ...models.schemas import PlannerOverrideRequest
from ...services.audit_service import audit_service

router = APIRouter(prefix="/decision", tags=["Decision & Audit"])

@router.get("/audit-log")
def get_audit_log():
    return audit_service.get_logs()

@router.post("/override")
def submit_planner_override(request: PlannerOverrideRequest):
    event = audit_service.record_event(
        event_type="PLANNER_OVERRIDE",
        actor=f"{request.planner_name} ({request.planner_role})",
        details=f"Overrode block timing to {request.new_start_time} ({request.new_duration_min}m). Reason: {request.reason}"
    )
    return {
        "status": "RECORDED",
        "audit_entry": event
    }
